From 887c2f04353a8952020ec738c6f2252d057d18c5 Mon Sep 17 00:00:00 2001
From: Rafael Palomar <rafael.palomar@ous-research.no>
Date: Tue, 28 Feb 2023 13:02:15 +0100
Subject: [PATCH 25/32] ENH: Make Testing subdirs subject to `BUILD_TESTING`

This makes QTCore and QTGUI `Testing` depend on the `BUILD_TESTING`
CMake variable.
---
 Base/QTCore/CMakeLists.txt | 14 ++++++++------
 Base/QTGUI/CMakeLists.txt  | 16 +++++++++-------
 2 files changed, 17 insertions(+), 13 deletions(-)

diff --git a/Base/QTCore/CMakeLists.txt b/Base/QTCore/CMakeLists.txt
index 86ee0e25f1..c1d406d2d3 100644
--- a/Base/QTCore/CMakeLists.txt
+++ b/Base/QTCore/CMakeLists.txt
@@ -292,9 +292,11 @@ endif()
 # Plugins
 #add_subdirectory(Plugins)
 
-# Testing
-# Note: Within both Testing/Cxx and Testing/Python, some code is useful to both Slicer and the
-#       Extensions. For that reason, the Testing directory is added inconditionnally.
-#       The code specific to Slicer will be excluded directly in either Testing/Python
-#       or Testing/Cxx.
-add_subdirectory(Testing)
+if(BUILD_TESTING)
+  # Testing
+  # Note: Within both Testing/Cxx and Testing/Python, some code is useful to both Slicer and the
+  #       Extensions. For that reason, the Testing directory is added inconditionnally.
+  #       The code specific to Slicer will be excluded directly in either Testing/Python
+  #       or Testing/Cxx.
+  add_subdirectory(Testing)
+endif()
diff --git a/Base/QTGUI/CMakeLists.txt b/Base/QTGUI/CMakeLists.txt
index 333b39c955..3db0072693 100644
--- a/Base/QTGUI/CMakeLists.txt
+++ b/Base/QTGUI/CMakeLists.txt
@@ -424,10 +424,12 @@ if(Slicer_BUILD_QT_DESIGNER_PLUGINS)
 endif()
 add_subdirectory(StylePlugins)
 
-#-----------------------------------------------------------------------------
-# Testing
-# Note: Within both Testing/Cxx and Testing/Python, some code is useful to both Slicer and the
-#       Extensions. For that reason, the Testing directory is added unconditionally.
-#       The code specific to Slicer will be excluded directly in either Testing/Python
-#       or Testing/Cxx.
-add_subdirectory(Testing)
+  #-----------------------------------------------------------------------------
+if(BUILD_TESTING)
+  # Testing
+  # Note: Within both Testing/Cxx and Testing/Python, some code is useful to both Slicer and the
+  #       Extensions. For that reason, the Testing directory is added unconditionally.
+  #       The code specific to Slicer will be excluded directly in either Testing/Python
+  #       or Testing/Cxx.
+  add_subdirectory(Testing)
+endif()
-- 
2.39.2

